pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/ComponentImpl.java

```diff
@@ -1147,16 +1075,17 @@ public Response triggerFunction(final String tenant,
         }
     }
 
-    public Response getFunctionState(final String tenant,
-                                     final String namespace,
-                                     final String functionName,
-                                     final String key) {
+    public FunctionState getFunctionState(final String tenant,
+                                          final String namespace,
+                                          final String functionName,
+                                          final String key) {
+
         if (!isWorkerServiceAvailable()) {
-            return getUnavailableResponse();
+            throwUnavailableException();
         }
 
         if (null == worker().getStateStoreAdminClient()) {
-            return getStateStoreUnvailableResponse();
+            getStateStoreUnvailableResponse();
```

Comment:
this should be a return statement

Reply:
We are throwing an exception why would there be a return value